Tests/AppTests/QueryPerformanceTests.swift

```diff
@@ -139,15 +139,22 @@ class QueryPerformanceTests: XCTestCase {
               r.summary,
               v.package_name,
               array_length(doc_archives, 1) >= 1 AS has_docs,
-              ARRAY(SELECT DISTINCT JSONB_OBJECT_KEYS(type) FROM products WHERE products.version_id = v.id) AS product_types,
+              ARRAY(
+                SELECT DISTINCT JSONB_OBJECT_KEYS(type) FROM products WHERE products.version_id = v.id
+                UNION
+                SELECT * FROM (
+                  SELECT DISTINCT JSONB_OBJECT_KEYS(type) AS "type" FROM targets
+                  WHERE targets.version_id = v.id) AS macro_targets
+                WHERE type = 'macro'
+              ) AS product_types,
               ARRAY(SELECT DISTINCT name FROM products WHERE products.version_id = v.id) AS product_names,
               TO_TSVECTOR(CONCAT_WS(' ', COALESCE(v.package_name, ''), r.name, COALESCE(r.summary, ''), ARRAY_TO_STRING(r.keywords, ' '))) AS tsvector
             FROM packages p
               JOIN repositories r ON r.package_id = p.id
               JOIN versions v ON v.package_id = p.id
             WHERE v.reference ->> 'branch' = r.default_branch
             """)
-        try await assertQueryPerformance(query, expectedCost: 31_300, variation: 500)
+        try await assertQueryPerformance(query, expectedCost: 55_000, variation: 500)
```

Comment:
That's a big increase in search view creation cost. The actual query time on `dev` is ~1s now. We run this every every 25-30s, so it's taking significant overall time.

I think it's still ok but we should reconsider doing this once we've gathered some experience with the feature or in case we make further extensions to search.

Reply:
Search view updates are linked to analysis passes. We could also reduce relative cost by not updating search on every analysis pass but on a "once every 5min" global timer.